addons/account/models/account_account.py

```diff
@@ -346,20 +402,160 @@ def _check_account_is_bank_journal_bank_account(self):
+            values = self.env['ir.property'].sudo().with_company(company)._get_multi('name', 'account.account', self.ids)
+            for record in self:
+                if (company in self.env.company.parent_ids | record.company_ids):
+                    record['name'] = values.get(record.id)
+            if all(record.name for record in self):
+                break
+
+    def _search_name(self, operator, value):
+        return self._search_company_dependent_field_fallback_on_parent('name', operator, value)
+
+    def _inverse_name(self):
+        values = {
+            record.id: self._fields['name'].convert_to_write(record['name'], record)
+            for record in self
+        }
+        for record in self:
+            for parent_company in (self.env.company.parent_ids - self.env.company):
+                name_for_parent = record.with_company(parent_company).name
+                if name_for_parent == record.name:
+                    values[record.id] = False
+                elif name_for_parent:
+                    break
+        self.env['ir.property'].sudo()._set_multi('name', 'account.account', values)
+
+    def _compute_code(self):
+        for company in self.env.company.parent_ids:
+            values = self.env['ir.property'].sudo().with_company(company)._get_multi('code', 'account.account', self.ids)
+            for record in self:
+                record['code'] = values.get(record.id)
+            if all(record.code for record in self):
+                break
+
+    def _search_code(self, operator, value):
+        return self._search_company_dependent_field_fallback_on_parent('code', operator, value)
+
+    def _inverse_code(self):
+        values = {
+            record.id: self._fields['code'].convert_to_write(record['code'], record)
+            for record in self
+        }
+        for record in self:
+            for parent_company in (self.env.company.parent_ids - self.env.company):
+                code_for_parent = record.with_company(parent_company).code
+                if code_for_parent == record.code:
+                    values[record.id] = False
+                elif code_for_parent:
+                    break
+        self.env['ir.property'].sudo()._set_multi('code', 'account.account', values)
+
+    def _search_company_dependent_field_fallback_on_parent(self, field_name, operator, value):
+        operator, value, negate = self.env['ir.property']._get_value_field_domain(field_name, 'account.account', operator, value)
+
+        field_id = self.env['ir.model.fields']._get('account.account', field_name).id
+        field_type = self.env['account.account']._fields[field_name].type
+        properties_yes = self.env['ir.property'].sudo().search([('fields_id', '=', field_id), (TYPE2FIELD[field_type], operator, value)])
+        properties_no = self.env['ir.property'].sudo().search([('fields_id', '=', field_id), '!', (TYPE2FIELD[field_type], operator, value)])
```

Comment:
Can't this be a performance issue?

Reply:
It's definitely going to be slower than what we had. 
How bad do you think it is? I was thinking that with a small enough number of accounts, these queries would perform pretty fast. 
Although, I'm surprised that there isn't any index on fields_id in ir_property. That might be a problem...

A small improvement for this query might be to search just on fields_id, and then do a filtered_domain to separate the properties that pass from those that don't pass.

Reply:
(oh, I just noticed that there actually is an index on fields_id since it's the first column of the multi-column unique index for the table)